src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,125 @@
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  setUserId: (userId: number) => void,
+};
+
+type State = {
+  currentInput: string,
+  statusTodo: string,
+  todos: Todo[],
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state:State = {
+    currentInput: '',
+    statusTodo: '',
+    todos: [],
+  };
+
+  componentDidMount() {
+    const loadTodosFromServer = async () => {
+      const todosFromServer = await getTodos();
+
+      this.setState({
+        todos: todosFromServer,
+      });
+    };
+
+    loadTodosFromServer();
+  }
+
+  handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      currentInput: event.target.value,
+    });
+  };
+
+  changeStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      statusTodo: event.target.value,
+    });
+  };
+
+  sortTodos = () => {
+    const { todos, currentInput, statusTodo } = this.state;
+    let copyTodos;
+
+    const lowerCaseCurrentInput = currentInput.toLowerCase();
+
+    switch (statusTodo) {
+      case 'active':
+        copyTodos = todos.filter((value) => (!value.completed));
+        break;
+      case 'completed':
+        copyTodos = todos.filter((value) => (value.completed));
+        break;
+      default:
+        copyTodos = todos;
+        break;
+    }
+
+    return copyTodos.filter((value) => {
+      if (value.title.toLowerCase().includes(lowerCaseCurrentInput)) {
+        return true;
+      }
+
+      return false;
+    });
+  };
+
+  render() {
+    const { setUserId } = this.props;
+    const todos = this.sortTodos();
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+        <input
+          type="text"
+          value={this.state.currentInput}
+          onChange={this.handleInput}
+        />
+        <select
+          value={this.state.statusTodo}
+          onChange={this.changeStatus}
+        >
+          <option value="">All</option>
+          <option value="active">Active</option>
+          <option value="completed">Completed </option>
+        </select>
+
+        <div className="TodoList__list-container">
+          <ul className="TodoList__list">
+            {todos.map((todo) => (
+              <li
+                key={todo.id}
+                className={classNames(
+                  'TodoList__item',
+                  { 'TodoList__item--unchecked': !todo.completed },
+                  { 'TodoList__item--checked': todo.completed },
```

Comment:
use one of these classes by default and set one with boolean value

Reply:
also, I don't see style files with these classes, add them

Reply:
maybe it'd make sense to make checkbox checked if todo is completed?
![image](https://user-images.githubusercontent.com/37161315/163772639-a48b9904-b2f2-4af9-825e-450c5cc0594e.png)